contracts/rbn-staking/FeeCustody.sol

```diff
@@ -0,0 +1,425 @@
+  // Address of fee distributor contract for RBN lockers to claim
+  IFeeDistributor public feeDistributor;
+
+  // % allocation (0 - 100%) from protocol revenue to allocate to RBN lockers.
+  // 2 decimals. ex: 10% = 1000
+  uint256 public pctAllocationForRBNLockers;
+
+  uint256 public constant TOTAL_PCT = 10000; // Equals 100%
+  ISwapRouter public constant UNIV3_SWAP_ROUTER =
+    ISwapRouter(0x68b3465833fb72A70ecDF485E0e4C7bD8665Fc45);
+
+  // Intermediary path asset for univ3 swaps.
+  // Empty if direct pool swap between asset and distribution asset
+  mapping(address => bytes) public intermediaryPath;
+
+  // Oracle between asset/usd pair for total
+  // reward approximation across all assets earned
+  mapping(address => address) public oracles;
+
+  address[1000] public assets;
+  // Index of empty slot in assets array
+  uint256 public lastAssetIdx;
+
+  // Keeper for weekly distributions
+  address public keeper;
+
+  // Events
+  event NewAsset(address asset, bytes intermediaryPath);
+  event RecoveredAsset(address asset);
+  event NewFeeDistributor(address feeDistributor);
+  event NewRBNLockerAllocation(uint256 pctAllocationForRBNLockers);
+  event NewDistributionToken(address distributionToken);
+  event NewProtocolRevenueRecipient(address protocolRevenueRecipient);
+  event NewKeeper(address keeper);
+
+  /**
+   * @notice
+   * Constructor
+   * @param _pctAllocationForRBNLockers percent allocated for RBN lockers (100% = 10000)
+   * @param _feeDistributor address of fee distributor where protocol revenue claimable
+   * @param _protocolRevenueRecipient address of multisig
+   * @param _admin admin
+   * @param _keeper keeper
+   */
+  constructor(
+    uint256 _pctAllocationForRBNLockers,
+    address _feeDistributor,
+    address _protocolRevenueRecipient,
+    address _admin,
+    address _keeper
+  ) {
+    require(_feeDistributor != address(0), "!_feeDistributor");
+    require(
+      _protocolRevenueRecipient != address(0),
+      "!_protocolRevenueRecipient"
+    );
+    require(_admin != address(0), "!_admin");
+    require(_keeper != address(0), "!_keeper");
+
+    pctAllocationForRBNLockers = _pctAllocationForRBNLockers;
+    feeDistributor = IFeeDistributor(_feeDistributor);
+    protocolRevenueRecipient = _protocolRevenueRecipient;
+    keeper = _keeper;
+  }
+
+  /**
+   * @notice
+   * Swaps RBN locker allocation of protocol revenu to distributionToken,
+   * sends the rest to the multisig
+   * @dev Can be called by keeper
+   * @param _minAmountOut min amount out for every asset type swap.
+   * will need to be in order of assets in assets[] array. should be
+   * fine if we keep track.
+   * @return toDistribute amount of distributionToken distributed to fee distributor
+   */
+  function distributeProtocolRevenue(uint256[] calldata _minAmountOut)
+    external
+    returns (uint256 toDistribute)
+  {
+    require(msg.sender == keeper, "!keeper");
+
+    if (address(distributionToken) == WETH) {
```

Comment:
How would this handle wstETH, given there are no liquid wstETH/USDC pools?

Reply:
oh, wow was under impression we were unwrapping to eth for some reason. updating now

Reply:
I changed it. But getting an "Transaction reverted: function selector was not recognized and there's no fallback nor receive function" on [Line 250](https://github.com/ribbon-finance/governance/blob/27280f161bf28166aa99b08b29ee6a29d9ac07d9/contracts/rbn-staking/FeeCustody.sol#L250) with exchange() when calling distributeProtocolRevenue(). Do you know why off-hand? I tried adding some getters from that contract like get_dy() or A() which all worked